fdbserver/QuietDatabase.actor.cpp

```diff
@@ -284,6 +284,52 @@ ACTOR Future<vector<WorkerInterface>> getStorageWorkers(Database cx,
 	return result;
 }
 
+int64_t extractMaxQueueSize(const std::vector<Future<TraceEventFields>>& messages,
+                            const std::vector<StorageServerInterface>& servers) {
+
+	int64_t maxQueueSize = 0;
+	UID maxQueueServer;
+
+	for (int i = 0; i < messages.size(); i++) {
+		try {
+			auto queueSize = getQueueSize(messages[i].get());
+			if (queueSize > maxQueueSize) {
+				maxQueueSize = queueSize;
+				maxQueueServer = servers[i].id();
+			}
+		} catch (Error& e) {
+			TraceEvent("QuietDatabaseFailure")
+			    .detail("Reason", "Failed to extract MaxStorageServerQueue")
+			    .detail("SS", servers[i].id());
+			for (auto& m : messages) {
+				TraceEvent("Messages").detail("Info", m.get().toString());
+			}
+			throw;
+		}
+	}
+
+	TraceEvent("QuietDatabaseGotMaxStorageServerQueueSize")
+	    .detail("Stage", "MaxComputed")
+	    .detail("Max", maxQueueSize)
+	    .detail("MaxQueueServer", format("%016" PRIx64, maxQueueServer.first()));
+	return maxQueueSize;
+}
+
+ACTOR Future<TraceEventFields> getStorageMetricsTimeout(UID storage, WorkerInterface wi) {
```

Comment:
FWIW, this seems like it could be done with some generic actors, such as

`map(orError(timeout(f)), [=](ErrorOr<TraceEventFields> e) {`...

at the call site but I'm not against the explicit version for readability.

Reply:
I started writing this, but it got quite a bit more complicated than what you wrote here. So I would prefer to leave it as is. Otherwise it might be short and unreadable.